sys/Makefile.dep

```diff
@@ -858,6 +857,13 @@ ifneq (,$(filter evtimer,$(USEMODULE)))
   endif
 endif
 
+ifneq (,$(filter xztimer_fallback,$(USEMODULE)))
+  USEMODULE := $(filter-out xztimer_fallback,$(USEMODULE))
+  ifeq (,$(filter ztimer_usec,$(USEMODULE)))
+    USEMODULE += xtimer
+  endif
+endif
```

Comment:
This kind of pattern is not used and is not a good idea to use in the build system, it is very very prone to dependency order issues.

Are we sure this is still worth it with `xtimer_ztimer_compat` being used by default now?

Reply:
This is a step to remove the xtimer support from these modules they should be easy conversion to ztimer. We may just remove the xtimer support from these modules entirely and skip this step.

I do not like writing these:

```
if there is ztimer
   do this 
elseif there is xtimer
   do that
```
Even reading this kind of code not nice since it distracts from what is the main purpose of that module.

Reply:
Then instead how about adapting the `coccinelle` scripts to have two versions a `ztimer_xtimer_compat` version and a `ztimer64_xtimer_compat` version, and simply spatch those files and directly make the conversion?

Reply:
#17892 has that done (without the coccinelle part)

Reply:
> This kind of pattern is not used and is not a good idea to use in the build system, it is very very prone to dependency order issues.

somehow i think the `xztimer_fallback` solves a dependency order issue for modules that support `ztimer` as well as `xtimer` through delaying the decision until it is necessary

If the first module in Makefile.dep supports x and z timer the dependancy will be decided at that point  and every later module will see the decision done there but in reality the module did not care. - may be this needs another PR